### DISCUSSION TITLE
V2 Usage

### DISCUSSION THREAD

**Discussion by nbonamy** [flyerhq/flutter_chat_ui#671] [Q&A] [answered]
Hey thanks for the great library. Using the v2 for a project and it is working great. Some minor issues I am facing but I have been able to handle:

- Building my own ChatInput so using inputBuilder. I am under the impression that for layout of the main screen to be properly calculated I have to use `ChatInputHeightNotifier` (in `_updateInputHeight`). But this is not "exported" so I need to `import 'package:flutter_chat_ui/src/utils/chat_input_height_notifier.dart';` for which of course I get a warning. If I don't handle `ChatInputHeightNotifier`, the chat area goes behind my custom `ChatInput`.

- I am also using a custom `ChatController` as I want to have my own data model for messages. In my custom `ChatController` I end up having to store all the `flutter_chat_ui` messages in case of update because of (I believe) https://github.com/flyerhq/flutter_chat_ui/blob/be518f82717d402fbae0e858021801af18f61dc9/packages/flutter_chat_ui/lib/src/chat_message/chat_message_internal.dart#L52. Maybe just checking `id` equality would be enough It will allow me to only save messages with my model and recreate `flutter_chat_ui` messages in `List<Message> get messages` on the fly.

Keep up the great work!

https://github.com/user-attachments/assets/4967cb8a-bf67-4c08-a792-c75cacf0da19

**Comment by demchenkoalex** [comment]
Hello!

So for the first problem it would be enough to export `ChatInputHeightNotifier`? I also plan to include option to have solid background input, without blur, so this would not be needed.

For the second issue I think I can reduce equality check to IDs only, will that help?

**Comment by nbonamy** [comment]
Yes both will be great!

**Comment by KylinShinlyLin** [comment]
Does copying from the bubbles lose newline characters after your Markdown is parsed?

**Comment by demchenkoalex** [comment, marked as answer]
released in [2.0.0-dev.2](https://github.com/flyerhq/flutter_chat_ui/releases/tag/2.0.0-dev.2)
